uint8 randNum
uint8 callback
